### PR TITLE
Add user management functionality for admins

### PR DESCRIPTION
**Pull request by IvaskevychYuriy**
Refactoring and cleanup

### CODE REVIEW

File: SimpleDrive.App/ClientApp/components/UserEditDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        try {
+            await usersService.update(this.state.editModel);
+            toast.info(`Updated user '${this.props.user.userName}'`)
+        } catch {
+            toast.error("Could not edit user");
+        }
+    }
+
+    private handleAllowQuotaChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        console.log(event.target.checked);
```
Comment (nazar554) [inline SimpleDrive.App/ClientApp/components/UserEditDialog.tsx]: remove `console.log`

File: SimpleDrive.App/ClientApp/components/UsersGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            this.setState({
+                users: this.state.users.filter(u => u.id !== user.id)
+            });
+        } catch {
+            toast.error(`Couldn't delete user '${user.userName}'`);
+        }
+    }
+
+    private editUser = (user: User) => {
+        console.log(user);
```
Comment (nazar554) [inline SimpleDrive.App/ClientApp/components/UsersGrid.tsx]: and here

File: SimpleDrive.App/ClientApp/components/UserEditDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+const permissionStyle: React.CSSProperties = {
+    flex: 1
+}
+
+export default class UserEditDialog extends React.Component<UserEditProps, UserEditState> {
+
+    constructor(props: UserEditProps) {
+        super(props);
+        
+        console.log(props.user);
```
Comment (nazar554) [inline SimpleDrive.App/ClientApp/components/UserEditDialog.tsx]: and here

File: SimpleDrive.App/ClientApp/components/UserEditDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        this.state = {
+            editModel: Object.assign({}, props.user) as UserEditModel
+        }; 
+    }
+
+    componentWillReceiveProps(props: UserEditProps) {  
+        console.log(props.user);
```
Comment (nazar554) [inline SimpleDrive.App/ClientApp/components/UserEditDialog.tsx]: and here

File: SimpleDrive.App/ClientApp/components/UsersGrid.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        this.setState({
+            currentEditingUser: user,
+            editingEnabled: true
+        });
+    }
+
+    private onEditClose = async () => {
+        this.setState({
+            users: await usersService.get(),
+            currentEditingUser: {} as User,
```
Comment (nazar554) [inline SimpleDrive.App/ClientApp/components/UsersGrid.tsx:64]: provide default anonymous user model?

File: SimpleDrive.App/ClientApp/components/UsersGrid.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    editingEnabled: boolean;
+}
+
+export default class UsersGrid extends React.Component<UsersGridProps, UsersGridState> {
+    constructor(props: UsersGridProps) {
+        super(props);
+
+        this.state = {
+            users: [],
+            currentEditingUser: {} as User,
```
Comment (nazar554) [inline SimpleDrive.App/ClientApp/components/UsersGrid.tsx:31]: provide default anonymous user model?

**Review by nazar554 — CHANGES_REQUESTED** [review]

File: SimpleDrive.App/ClientApp/components/UserEditDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        this.setState({
+            editModel: Object.assign({}, this.state.editModel, {
+                quotaAllowed: event.target.checked ? 0 : null
+            })
+        });
+    }
+
+    private handleQuotaChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        this.setState({
+            editModel: Object.assign({}, this.state.editModel, {
```
Comment (nazar554) [inline SimpleDrive.App/ClientApp/components/UserEditDialog.tsx]: spread

File: SimpleDrive.App/ClientApp/components/UserEditDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+            toast.info(`Updated user '${this.props.user.userName}'`)
+        } catch {
+            toast.error("Could not edit user");
+        }
+    }
+
+    private handleAllowQuotaChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        console.log(event.target.checked);
+        this.setState({
+            editModel: Object.assign({}, this.state.editModel, {
```
Comment (nazar554) [inline SimpleDrive.App/ClientApp/components/UserEditDialog.tsx]: spread

File: SimpleDrive.App/ClientApp/components/UserEditDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        console.log(props.user);
+        this.state = {
+            editModel: Object.assign({}, props.user) as UserEditModel
+        }; 
+    }
+
+    componentWillReceiveProps(props: UserEditProps) {  
+        console.log(props.user);  
+        this.setState({
+            editModel: Object.assign({}, props.user) as UserEditModel
```
Comment (nazar554) [inline SimpleDrive.App/ClientApp/components/UserEditDialog.tsx]: spread

File: SimpleDrive.App/ClientApp/components/UserEditDialog.tsx [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+export default class UserEditDialog extends React.Component<UserEditProps, UserEditState> {
+
+    constructor(props: UserEditProps) {
+        super(props);
+        
+        console.log(props.user);
+        this.state = {
+            editModel: Object.assign({}, props.user) as UserEditModel
```
Comment (nazar554) [inline SimpleDrive.App/ClientApp/components/UserEditDialog.tsx]: spread

**Review by nazar554 — CHANGES_REQUESTED** [review]

**Comment by IvaskevychYuriy** [comment]
Fixed code review errors

**Pull request merged**